tokio/src/runtime/basic_scheduler.rs

```diff
@@ -214,12 +282,15 @@ where
     CURRENT.set(context, || f(scheduler, context))
 }
 
-impl<P> Drop for BasicScheduler<P>
-where
-    P: Park,
-{
+impl<P: Park> Drop for BasicScheduler<P> {
     fn drop(&mut self) {
-        enter(self, |scheduler, context| {
+        let mut inner = {
+            let mut lock = self.inner.lock().expect("BasicScheduler Inner lock");
+            lock.take()
+                .expect("Oh no! We never placed the Inner state back!")
+        };
```

Comment:
nit, tioli: can't this be
```suggestion
    let mut inner = self
        .inner
        .lock()
        .expect("BasicScheduler Inner lock")
        .take()
        .expect("Oh no! We never placed the Inner state back!");
```